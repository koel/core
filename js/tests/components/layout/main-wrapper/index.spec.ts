import Component from '@/components/layout/main-wrapper/index.vue'
import { shallow } from '@/tests/adapter'

describe('component/layout/main-wrapper/index', () => {
  it('renders properly', async () => {
    const wrapper = shallow(Component)
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
})
