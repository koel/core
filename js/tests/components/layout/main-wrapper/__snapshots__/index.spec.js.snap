// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/layout/main-wrapper/index renders properly 1`] = `
<div id="mainWrapper">
  <sidebar-stub></sidebar-stub>
  <section id="mainContent">
    <!---->
    <!---->
    <section id="homeWrapper">
      <h1 class="heading"><span>Oh hai!</span></h1>
      <div class="main-scroll-wrap">
        <div class="two-cols">
          <section style="display: none;">
            <h1>Most Played</h1>
            <ol class="top-song-list"></ol>
          </section>
          <section class="recent">
            <h1>Recently Played
              <!---->
            </h1>
            <ol class="recent-song-list" style="display: none;"></ol>
            <p class="none">
              Your recently played songs will be displayed here.<br>
              Start listening!
            </p>
          </section>
        </div>
        <section class="recently-added" style="display: none;">
          <h1>Recently Added</h1>
          <div class="two-cols">
            <div class="wrapper as-list"></div>
            <div>
              <ul class="recently-added-song-list" style="display: none;"></ul>
            </div>
          </div>
        </section>
        <section class="top-artists" style="display: none;">
          <h1>Top Artists</h1>
          <div class="wrapper as-null"></div>
        </section>
        <section class="top-albums" style="display: none;">
          <h1>Top Albums</h1>
          <div class="wrapper as-null"></div>
        </section>
        <!---->
      </div>
    </section>
    <section id="queueWrapper" style="display: none;">
      <h1 class="heading"><span title="That's a lot of U's and E's">Current Queue
      <!----> <span class="meta" style="display: none;">
        0 songs
        •
        00:00
      </span></span>
        <!---->
      </h1>
      <!---->
      <div class="none">
        <p>Empty spaces. Abandoned places.</p>
        <p>How about
          <a class="start">shuffling all songs</a>?
        </p>
      </div>
    </section>
    <section id="songsWrapper" style="display: none;">
      <h1 class="heading"><span>All Songs
      <!----> <span class="meta" style="display: none;">
        0 songs
        •
        00:00
      </span></span>
        <!---->
      </h1>
      <!---->
    </section>
    <section id="albumsWrapper" style="display: none;">
      <h1 class="heading"><span>Albums</span>
        <!---->
      </h1>
      <div class="albums main-scroll-wrap as-">
        <!---->
      </div>
    </section>
    <section id="artistsWrapper" style="display: none;">
      <h1 class="heading"><span>Artists</span>
        <!---->
      </h1>
      <div class="artists main-scroll-wrap as-">
        <!---->
      </div>
    </section>
    <section id="playlistWrapper" style="display: none;">
      <!---->
    </section>
    <section id="favoritesWrapper" style="display: none;">
      <h1 class="heading"><span>Songs You Love
      <!----> <span class="meta" style="display: none;">
        0 songs
        •
        00:00
        <!----></span></span>
        <!---->
      </h1>
      <!---->
      <div class="none">
        Start loving!
        Click the <i class="fa fa-heart" style="margin: 0px 5px;"></i> icon when a song is playing to add it
        to this list.
      </div>
    </section>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </section>
  <section id="extra" class="showing">
    <div class="tabs">
      <div role="tablist" class="clear"><button aria-selected="true" aria-controls="extraPanelLyrics" id="extraTabLyrics" role="tab">
          Lyrics
        </button> <button aria-controls="extraPanelArtist" id="extraTabArtist" role="tab">
          Artist
        </button> <button aria-controls="extraPanelAlbum" id="extraTabAlbum" role="tab">
          Album
        </button>
        <!---->
      </div>
      <div class="panes">
        <div aria-labelledby="extraTabLyrics" id="extraPanelLyrics" role="tabpanel" tabindex="0">
          <!---->
        </div>
        <div aria-labelledby="extraTabArtist" id="extraPanelArtist" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
        <div aria-labelledby="extraTabAlbum" id="extraPanelAlbum" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
        <div aria-labelledby="extraTabAlbum" id="extraPanelAlbum" role="tabpanel" tabindex="0" style="display: none;">
          <!---->
        </div>
      </div>
    </div>
  </section>
  <div class="modal-wrapper">
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;
